backend/src/main/java/com/roc/app/user/general/UserController.java

```diff
@@ -0,0 +1,39 @@
+package com.roc.app.user.general;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import com.roc.app.user.general.exception.UserNotFoundException;
+
+@Controller
+@RequestMapping("/api/users")
+public class UserController {
+
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping("/id/{email}")
+    public ResponseEntity<Long> getUserIdByEmail(@PathVariable String email) {
+        try {
+            Long userId = userService.getUserIdByEmail(email);
+            return ResponseEntity.ok(userId);
+        } catch (UserNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+    }
```

Comment:
Since this is used in the context of the logged-in user, passing the email isn't necessary and may pose a security risk. It's better to fetch the ID directly from the authenticated user:
```suggestion
    @GetMapping("/id")
    public ResponseEntity<Long> getUserId(Authentication authentication) {
        User user = (User) authentication.getPrincipal();
        return ResponseEntity.ok(user.getUserId());
    }
```
Keeps the API simpler and avoids exposing user data unnecessarily.